tests/component-plugin/src/main/java/org/mule/functional/api/flow/FlowRunner.java

```diff
@@ -221,9 +230,15 @@ private ExecutionCallback<CoreEvent> getFlowRunCallback() {
       CoreEvent result;
       try {
         result = Flux.<CoreEvent>create(fluxSink -> {
-          fluxSink.next(getOrBuildEvent());
+          CoreEvent event = getOrBuildEvent();
+          fluxSink.next(event);
           sinkReference.set(fluxSink);
-        }).transform(flow::apply).blockFirst();
+        }).doOnNext(event -> privilegedProfilingService
+            .ifPresent(privilegedProfilingService -> privilegedProfilingService.startComponentSpan(event, flow)))
+            .transform(flow::apply)
+            .doOnNext(event -> privilegedProfilingService
+                .ifPresent(privilegedProfilingService -> privilegedProfilingService.endComponentSpan(event)))
```

Comment:
If the flow fails, the span is never ended, maybe a doOnError should be added?

Reply:
This is for capturing the spans in a test environment only. I think it will be correct for the span not to end.

Reply:
I've change this to end the span in case an error happens.